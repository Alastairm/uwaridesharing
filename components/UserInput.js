import React, { Component, Proptypes } from 'react';
import { StyleSheet, Text, TextInput } from 'react-native';

const styles = StyleSheet.create({
  invalid: {
    borderRadius: 5,
    borderColor: 'red',
  },
  valid: {
    borderRadius: 1,
    borderColor: 'black'
  }
})

class UserInput extends Component {
  constructor(props) {
    super(props);
    this.state = {text: "", style:{}}
    this.inputProps = this.inputProps.bind(this);
    this.onChange = this.onChange.bind(this)
  }
  inputProps() {
    const { field, labels, placeholders, keyboardTypes } = this.props;
    return {
      label: labels[field],
      placeholder: placeholders[field],
      keyboardType: keyboardTypes[field],
    }
  }
  onChange(text) {
    this.setState({text});
    this.validate();
    this.props.onChange(this.props.field, text);
  }
  validate() {
    // if (this.state.field == "email") {
      if (!emailRegex.test(this.state.text)) {
        this.setState({invalid: true});
      }
    // }
  }
  render() {
    const { label, placeholder, keyboardType, onChange } = this.inputProps();
    return (
      <TextInput
        style = {this.state.valid ? styles.valid: styles.invalid}
        value={this.state.text}
        onChangeText={this.onChange}
        keyboardType={keyboardType}
        placeholder={placeholder}
      />
    );
  }
}
// UserInput.propTypes = {
//   field: Proptypes.string,
//   onChange: Proptypes.func,
// }
UserInput.defaultProps = {
  labels: {
    name: "NAME",
    email: "EMAIL",
    phone: "MOBILE",
  },
  placeholders: {
    name: "Your Name",
    email: "example@uwa.edu.au",
    phone: "04 1234 5678",
  },
  keyboardTypes: {
    name: "default",
    email: "email-address",
    phone: "phone-pad",
  }
}

export default UserInput;

  const emailRegex = re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@(uwa.edu.au|student.uwa.edu.au)$/;
