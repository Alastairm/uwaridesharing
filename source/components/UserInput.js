import React, { Component, PropTypes } from 'react';
import { StyleSheet, Text, TextInput } from 'react-native';

const styles = StyleSheet.create({
  valid: {
    backgroundColor: '#ffffff',
  },
  invalid: {
    backgroundColor: '#ff0000',
  },
})

export default class UserInput extends Component {
  static defaultProps = {
    labels: {
      name: "NAME",
      email: "EMAIL",
      phone: "MOBILE",
    },
    placeholders: {
      name: "Your Name",
      email: "example@uwa.edu.au",
      phone: "04 1234 5678",
    },
    keyboardTypes: {
      name: "default",
      email: "email-address",
      phone: "phone-pad",
    }
  };
  static PropTypes = {
    field: PropTypes.string.isRequired,
  };
  constructor(props) {
    super(props);
    this.state = {text: "", valid: true};
    this.inputProps = this.inputProps.bind(this);
    this.onChange = this.onChange.bind(this)
  }
  inputProps() {
    const { field, labels, placeholders, keyboardTypes } = this.props;
    return {
      label: labels[field],
      placeholder: placeholders[field],
      keyboardType: keyboardTypes[field],
    }
  }
  onChange(text) {
    this.setState({text: text});
    this.validate(text);
    this.props.onChange(this.props.field, text);
  }
  validate(text) {
     if (this.props.field == "email") {
      if (emailRegex.test(text)==true) {
        this.setState({valid: true});
      } else {
        this.setState({valid: false})
      }
     }
  }
  render() {
    const { label, placeholder, keyboardType, onChange } = this.inputProps();
    return (
      <TextInput
        style = {this.state.valid ? styles.valid: styles.invalid}
        value={this.state.text}
        onChangeText={this.onChange}
        keyboardType={keyboardType}
        placeholder={placeholder}
      />
    );
  }
}


const emailRegex = re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@(uwa.edu.au|student.uwa.edu.au)$/;
