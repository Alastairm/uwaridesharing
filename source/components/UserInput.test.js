import React from 'react';
import renderer from 'react-test-renderer';
import UserInput from './UserInput.js'
it('Name field matches previous snapshot', () => {
  const component = renderer.create(
    <UserInput field="name"/>
  ).toJSON();
  expect(component).toMatchSnapshot();
});
it('Email field matches previous snapshot', () => {
  const component = renderer.create(
    <UserInput field="email"/>
  ).toJSON();
  expect(component).toMatchSnapshot();
});
it('Phone field matches previous snapshot', () => {
  const component = renderer.create(
    <UserInput field="phone"/>
  ).toJSON();
  expect(component).toMatchSnapshot();
});
