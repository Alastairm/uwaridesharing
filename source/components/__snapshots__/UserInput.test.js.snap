exports[`test Email field matches previous snapshot 1`] = `
<TextInput
  keyboardType="email-address"
  onChangeText={[Function]}
  placeholder="example@uwa.edu.au"
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
  value="" />
`;

exports[`test Name field matches previous snapshot 1`] = `
<TextInput
  keyboardType="default"
  onChangeText={[Function]}
  placeholder="Your Name"
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
  value="" />
`;

exports[`test Phone field matches previous snapshot 1`] = `
<TextInput
  keyboardType="phone-pad"
  onChangeText={[Function]}
  placeholder="04 1234 5678"
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
  value="" />
`;
